### DISCUSSION TITLE
Automatically update flex recipes with GitHub Actions

### DISCUSSION THREAD

**Discussion by ebitkov** [symfony/symfony#57517] [Q&A] [answered]
Hello,

I recently looked into GitHubs Dependabot and I thought: Why not upgrade Flex recipes automatically with GitHub actions in a similar fashion: Check for update, update on a new branch and create a pull request.

There is no way nobody had the same idea already, I thought, but to my surprise I couldn't find anything about it - not even a post or article theorising about it.

So I wonder, is it even desirable to update flex recipes automatically? 

In my experience it's rarely a problem, if a recipe is outdated for some time and upgrading them occasionally, mostly when updating core components and bundles is good enough.

And sometimes it's even better to not upgrade immediately, as the [latest discussion about entity auto-mapping in controllers has shown](https://github.com/symfony/recipes/pull/1302).

But it's still a bit of a hassle the keep updated about changes in the recipe repo and remember to actually run the script. Especially if you maintain 10-20 projects, touching some of them only every few months.

Technically it seems not that difficult to implement such an action - run the composer script, create a new commit on a new branch, open a pull request. The user decides to merge and resolves conflicts.

So, am I the only one, who thinks that's a great idea? Or are there any technical problems I'm not wise enough to see? Looking forward to your thoughts.

**Comment by mbabker** [comment, marked as answer]
> So I wonder, is it even desirable to update flex recipes automatically?

Personally, I'd say no.  Recipes are writing files to your application, and may be later modified by you.  While it's true a number of the recipe-generated files might not be touched at all, they're still application-level configuration and I'm not sure I'm comfortable with these files being modified by a Dependabot-like system.

IMO they're great for starting up your app development, or integrating new features from new packages, and they can be a good reference for what the suggested defaults are as things change.  But, I don't think recipes need to be kept up-to-date in the same way that you would inspect your `composer outdated` or `npm outdated` lists and update those dependencies.

**Comment by ebitkov** [comment]
So, recipes are more of a boilerplate, than really something you need to be on top of? And it's enough to just upgrade them whenever you jump to a new minor/major version? Therefore a auto-update action isn't really that required, hence nobody ever created one. 

Makes sense.
